PWGHF/TableProducer/candidateSelectorXicToPKPi.cxx

```diff
@@ -239,15 +290,45 @@ struct HfCandidateSelectorXicToPKPi {
 
       if (pidXicToPKPi == 0 && pidXicToPiKP == 0) {
         hfSelXicToPKPiCandidate(statusXicToPKPi, statusXicToPiKP);
+        if (applyMl) {
+          hfMlXicToPKPiCandidate(outputMlXicToPKPi, outputMlXicToPiKP);
+        }
         continue;
       }
-
       if ((pidXicToPKPi == -1 || pidXicToPKPi == 1) && topolXicToPKPi) {
         statusXicToPKPi = 1; // identified as Xic->pKpi
       }
       if ((pidXicToPiKP == -1 || pidXicToPiKP == 1) && topolXicToPiKP) {
         statusXicToPiKP = 1; // identified as Xic->piKp
       }
+      SETBIT(statusXicToPKPi, aod::SelectionStep::RecoPID);
+      SETBIT(statusXicToPiKP, aod::SelectionStep::RecoPID);
```

Comment:
Same as above. Moreover, at this point you are overwriting the assignments made in L299 and L302.